streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -83,9 +85,45 @@ public void register(final StateRestorer restorer) {
 
         final Set<TopicPartition> restoringPartitions = new HashSet<>(needsRestoring.keySet());
         try {
-            final ConsumerRecords<byte[], byte[]> allRecords = poll(restoreConsumer, 10);
+            int totalNumberOfRecords = 0;
+            final Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> allRecords = new HashMap<>();
+            while (totalNumberOfRecords < DEFAULT_MAX) {
```

Comment:
I am not sure why we need `DEFAULT_MAX` here? Can't we just restore whatever `poll(restoreConsumer, 10)` returns?

Reply:
Bill pointed out above that we do not want to restore too many records since it would cause too much latency. Also, in KAFKA-6730's description, it said to "to just consuming until `poll()` does not return any data". However, there is a chance this condition could not be fulfilled, particularly since poll() can continue to return records indefinitely, then we will terminate once we hit this ```DEFAULT_MAX``` parameter to ensure that the number of records restored will be relatively small.

Reply:
@ConcurrencyPractitioner thanks for updating the PR.   My point from before was that we should restore each batch of records returned from each `poll()` call vs. keeping all returned records in memory and start the restore process when there no more records to fetch.  Sorry if I did not make that point very clear.